### PR TITLE
ENH: tracking_quality function

### PR DESCRIPTION
**Pull request by hong2223**
Two enhancements for the `tracking_quality()`/`_split_overlaps()` function:
- previously `_split_overlaps` split records that span several days in midnight, e.g., a record that span 2 days is split into _started_at - 23:59:59_ and _00:00:00 - finished_at_, which means 1 second is lost whenever we perform a split. Now this is changed into _started_at - 00:00:00_ and _00:00:00 - finished_at_. Tests are changed accordingly.
- previously `_split_overlaps` did not deal with `duration` column. Now, this column is updated in the function.

### CODE REVIEW

File: tests/analysis/test_tracking_quality.py [hunk 1]
```diff
@@ -203,39 +230,32 @@ def test_split_overlaps_hours_case2(self, testdata_stps_tpls_geolife_long):
...
 
-    def test_accessors(self):
-        """The result obtained from model accessors should be the same as calling the function."""
-        pfs, _ = ti.io.dataset_reader.read_geolife(os.path.join("tests", "data", "geolife_long"))
-        pfs, stps = pfs.as_positionfixes.generate_staypoints(method="sliding", dist_threshold=25, time_threshold=5)
-        stps = stps.as_staypoints.create_activity_flag(time_threshold=15)
-        pfs, tpls = pfs.as_positionfixes.generate_triplegs(stps, method="between_staypoints")
-        stps, tpls, trips = ti.preprocessing.triplegs.generate_trips(stps, tpls, gap_threshold=15)
+    def test_split_overlaps_duration(self, testdata_stps_tpls_geolife_long):
+        """Test if _split_overlaps() function can split records that span several days."""
```
Comment (henrymartin1) [inline tests/analysis/test_tracking_quality.py:248]: I think the docstring is wrong

**Review by henrymartin1 — CHANGES_REQUESTED** [review]
Nice fix. I have some minor comments to address.

File: tests/analysis/test_tracking_quality.py [hunk 2]
```diff
@@ -161,6 +161,33 @@ def test_tracking_quality_user_error(self, testdata_stps_tpls_geolife_long):
         with pytest.raises(AttributeError):
             ti.analysis.tracking_quality._get_tracking_quality_user(user_0, granularity="random")
 
+    def test_accessors(self):
```
[the remark this answers is not in the document]
Comment (hong2223) [reply tests/analysis/test_tracking_quality.py]: the accessor testing is splitted into 3 functions

**Review by henrymartin1 — APPROVED** [review]
Great work, sorry that I missed this one.

**Pull request merged**